### PR TITLE
Moved the metadata into `setup.cfg`.

### PR DESCRIPTION
**Pull request by KOLANICH**
Added `pyproject.toml`

**Checklist:**

-   [x] If you are new to Scapy: I have checked [CONTRIBUTING.md](https://github.com/secdev/scapy/blob/master/CONTRIBUTING.md) (esp. section submitting-pull-requests)
-   [x] I squashed commits belonging together
-   [x] I added unit tests or explained why they are not relevant - just package building
-   [ ] I executed the regression tests for Python2 and Python3 (using `tox` or, `cd test && ./run_tests_py2, cd test && ./run_tests_py3`) - not needed
-   [x] If the PR is still not finished, please create a [Draft Pull Request](https://github.blog/2019-02-14-introducing-draft-pull-requests/)

Moves the metadata into the declarative config according to `PEP 621`, which is more editable and readable with automatic tools and somewhen in future will allow to forbid `setup.py` as an unjustified potential security threat.

### CODE REVIEW

File: setup.cfg [hunk 1]
```diff
@@ -2,8 +2,74 @@
...
+author = Philippe BIONDI
+author_email = phil(at)secdev.org
+maintainer = Pierre LALET, Gabriel POTTER, Guillaume VALADON
+license = GPL-2.0-only
 license_file = LICENSE
+description = Scapy: interactive packet manipulation tool
+keywords = network
+url = https://scapy.net
+download_url = https://github.com/secdev/scapy/tarball/master
+long_description = file: README.md
```
Comment (gpotter2) [inline setup.cfg:16]: Isn't this incompatible with the fact that we have a `setup(long_description=...)` in `setup.py` still?

**Review by gpotter2 — COMMENTED** [review]
Thanks for the PR.
I like modern stuff so I quite like it. Regarding compatibility: https://github.com/viraptor/arpy/pull/18#issuecomment-1152878275 (we still support Python 2.7...)

**Comment by KOLANICH** [comment]
>Moves the metadata into the declarative config according to PEP 621, 

Oops, forgot to add it. But given

>(we still support Python 2.7...)

Should I land the second commit converting `setup.cfg` into `pyproject.toml` too? It will make the wheel unbuildable on python 2 (the wheel still should be installable and workable, since the code inside supports py2 and the metadata in wheel says it does), but python 2 ... has been officially dropped for several years and BTW there were no real reasons to use python 2 even in 2013. So, I think, if one uses python 2, it should be his personal issue that he cannot build a wheel and has to use workarounds like fetching prebuilt wheels from pypi, unzipping it, replacing the sources and zipping them back. As long as these kind of workarounds work, we can consider python 2 support not dropped.

**Comment by gpotter2** [comment]
We are currently looking towards releasing Scapy 2.5.0 which should be the last version to support Python 2.7 (at least I'd like so).
We'll wait for that before breaking building on 2.7

**Comment by KOLANICH** [comment]
OK. In the current state this PR should support 2.7.

**Comment by polybassa** [comment]
LGTM

**Pull request closed without merging**